File: FizzBuzz/FizzBuzz.rb
class FizzBuzz
    @@FIZZ = 3
    @@BUZZ = 5

    def fizzbuzz(num)
        for i in 1..100 do
            if isFizz(i) && isBuzz(i) then
                print("FizzBuzz")
            elsif isFizz(i) then
                print("Fizz")
            elsif isBuzz(i) then
                print("Buzz")
            else
                print(i)
            end
            print("\n")
        end
    end

    private

    def isFizz(num)
        return num % @@FIZZ == 0
    end

    def isBuzz(num)
        return num % @@BUZZ == 0
    end

end

fz = FizzBuzz.new
fz.fizzbuzz(100)

File: FizzBuzz/FizzBuzz.rs
#[derive (Clone, Copy)]
struct FizzBuzz {
    fizz: i8,
    buzz: i8,
}

fn is_fizz(fz: FizzBuzz, num: i8) -> bool {
    return (num % fz.fizz == 0) as bool;
}

fn is_buzz(fz: FizzBuzz, num: i8) -> bool {
    return (num % fz.buzz == 0) as bool;
}

fn fizz_buzz(fz: FizzBuzz, num: i8) {
    for i in 1..num + 1 {
        if is_fizz(fz, i) && is_buzz(fz, i) {
            println!("FizzBuzz");
        } else if is_fizz(fz, i) {
            println!("Fizz");
        } else if is_buzz(fz, i) {
            println!("Buzz");
        } else {
            println!("{}", i);
        }
    }
}

fn main(){
    fizz_buzz(FizzBuzz{fizz:3, buzz:5}, 100);
}